src/cargo/util/toml/mod.rs

```diff
@@ -1123,7 +1132,13 @@ impl TomlManifest {
             project.resolver.as_ref(),
             me.workspace.as_ref().and_then(|ws| ws.resolver.as_ref()),
         ) {
-            (None, None) => None,
+            (None, None) => {
+                if edition == Edition::Edition2021 {
+                    Some(ResolveBehavior::V2)
+                } else {
+                    None
+                }
+            }
```

Comment:
I sort of forget the implications of this, can you remind me? What I'm worried about is a set of workspace crates that have mixed editions, but you could do that today as well with a set of crates in a workspace using different `resolver` versions too. How does Cargo handle that?

Reply:
The resolver is always global for a workspace, so if there is some workspace member that specifies `resolver="2"`, that is ignored (only the top-level matters).

However, thanks for pointing this out.  There was a bug where if you set 2021 in a member, it would warn about the resolver being set incorrectly.  I pushed a change where the default resolver behavior is now set in one place.